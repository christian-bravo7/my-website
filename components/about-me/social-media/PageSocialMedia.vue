<template>
  <section class="fixed z-40 left-0 grid gap-3 grid-cols-1 page-social-icons">
    <SocialIcon
      v-for="({ link, component }, index) in socialMediaIcons"
      :key="index"
      :link="link"
    >
      <Component :is="component" />
    </SocialIcon>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import { SocialMediaIcon } from '@/components/about-me/social-media/types';

import CodepenIcon from '~/assets/img/codepen.svg?inline';
import GithubIcon from '~/assets/img/github.svg?inline';
import InstagramIcon from '~/assets/img/instagram.svg?inline';
import LinkedinIcon from '~/assets/img/linkedin.svg?inline';

@Component({
  components: {
    CodepenIcon,
    GithubIcon,
    InstagramIcon,
    LinkedinIcon,
  },
})
export default class AppSocialIcons extends Vue {
  socialMediaIcons: Array<SocialMediaIcon> = [
    { link: 'https://codepen.io/christian-bravo', component: CodepenIcon },
    { link: 'https://github.com/christian-bravo7', component: GithubIcon },
    { link: 'https://www.instagram.com/christian.bravo_/', component: InstagramIcon },
    { link: 'https://www.linkedin.com/in/christian151-bravo/', component: LinkedinIcon },
  ]
}
</script>

<style lang="scss" scoped>
.page-social-icons {
  bottom: 4px;
  left: 4px;
}

@screen md {
  .page-social-icons {
    bottom: 12px;
    left: 12px;
  }
}
</style>
