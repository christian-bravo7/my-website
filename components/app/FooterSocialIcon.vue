<template>
  <a
    target="_blank"
    :href="link"
    class="footer-social-icon"
  >
    <slot />
  </a>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class FooterSocialIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly link!: string;
}
</script>

<style lang="scss" scoped>
.footer-social-icon {
  width: auto;

  > svg {
    transition: fill 250ms;
    fill: var(--footer-text);
  }

  &:hover {
    > svg {
      fill: var(--background);
    }
  }
}
</style>
