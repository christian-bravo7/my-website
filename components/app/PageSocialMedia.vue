<template>
  <section class="fixed z-50 left-0 app-social-icons grid gap-3 grid-cols-1">
    <FooterSocialIcon
      link="https://codepen.io/christian-bravo"
    >
      <CodepenIcon />
    </FooterSocialIcon>
    <FooterSocialIcon
      link="https://github.com/christian-bravo7"
    >
      <GithubIcon />
    </FooterSocialIcon>
    <FooterSocialIcon
      link="https://www.instagram.com/christian.bravo_/"
    >
      <InstagramIcon />
    </FooterSocialIcon>
    <FooterSocialIcon
      link="https://www.linkedin.com/in/christian151-bravo/"
    >
      <LinkedinIcon />
    </FooterSocialIcon>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import CodepenIcon from '~/assets/img/codepen.svg?inline';
import GithubIcon from '~/assets/img/github.svg?inline';
import InstagramIcon from '~/assets/img/instagram.svg?inline';
import LinkedinIcon from '~/assets/img/linkedin.svg?inline';

@Component({
  components: {
    CodepenIcon,
    GithubIcon,
    InstagramIcon,
    LinkedinIcon,
  },
})
export default class AppSocialIcons extends Vue {}

</script>

<style lang="scss" scoped>
.app-social-icons {
  position: fixed;
  bottom: rem(16);
  left: rem(16);
}
</style>
