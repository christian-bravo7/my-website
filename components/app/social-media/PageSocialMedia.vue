<template>
  <section class="fixed z-40 left-0 grid gap-3 grid-cols-1 page-social-icons">
    <SocialIcon
      v-for="({ link, component }, index) in socialLinks"
      :key="index"
      :link="link"
    >
      <Component :is="component" />
    </SocialIcon>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import CodepenIcon from '~/assets/img/codepen.svg?inline';
import GithubIcon from '~/assets/img/github.svg?inline';
import InstagramIcon from '~/assets/img/instagram.svg?inline';
import LinkedinIcon from '~/assets/img/linkedin.svg?inline';

@Component({
  components: {
    CodepenIcon,
    GithubIcon,
    InstagramIcon,
    LinkedinIcon,
  },
})
export default class AppSocialIcons extends Vue {
  socialLinks: Array<{ link: string, component: any }> = [
    { link: 'https://codepen.io/christian-bravo', component: CodepenIcon },
    { link: 'https://github.com/christian-bravo7', component: GithubIcon },
    { link: 'https://www.instagram.com/christian.bravo_/', component: InstagramIcon },
    { link: 'https://www.linkedin.com/in/christian151-bravo/', component: LinkedinIcon },
  ]
}
</script>

<style lang="scss" scoped>
/* stylelint-disable */

.page-social-icons {
  bottom: rem(4);
  left: rem(4);
}

@screen md {
  .page-social-icons {
    bottom: rem(12);
    left: rem(12);
  }
}
</style>
