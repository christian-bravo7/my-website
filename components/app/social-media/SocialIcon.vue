<template>
  <a
    target="_blank"
    :href="link"
    class="w-6 h-6 md:w-10 md:h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"
  >
    <slot />
  </a>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SocialIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly link!: string;
}
</script>

<style lang="scss" scoped>
/* stylelint-disable */

$light-color: #C5075D;
$dark-color: #5171FF;

.social-icon {
  padding: rem(4);

  &:hover {
    box-shadow: 0 0 0 5px rgba($light-color, 0.2);
  }

  svg {
    @apply fill-current;
  }
}

@screen md {
  .social-icon {
    padding: rem(10);
  }
}

html.dark-mode {
  .social-icon {
    &:hover {
      box-shadow: 0 0 0 5px rgba($dark-color, 0.2);
    }
  }
}

</style>
