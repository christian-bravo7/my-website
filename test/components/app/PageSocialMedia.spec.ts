import PageSocialMedia from '@/components/app/PageSocialMedia.vue';
import FooterSocialIcon from '@/components/app/FooterSocialIcon.vue';
import { mount } from '@vue/test-utils';

describe('PageSocialMedia component', () => {
  it('should render correclty', () => {
    const wrapper = mount(PageSocialMedia, {
      components: {
        FooterSocialIcon,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display 4 social icons', () => {
    const wrapper = mount(PageSocialMedia, {
      components: {
        FooterSocialIcon,
      },
    });

    const socialIcons = wrapper.findAllComponents(FooterSocialIcon);

    expect(socialIcons.length).toBe(4);
  });
});
