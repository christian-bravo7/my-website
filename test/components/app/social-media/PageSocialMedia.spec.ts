import PageSocialMedia from '@/components/app/social-media/PageSocialMedia.vue';
import SocialIcon from '@/components/app/social-media/SocialIcon.vue';
import { mount } from '@vue/test-utils';

describe('PageSocialMedia component', () => {
  it('should render correclty', () => {
    const wrapper = mount(PageSocialMedia, {
      components: {
        SocialIcon,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display 4 social icons', () => {
    const wrapper = mount(PageSocialMedia, {
      components: {
        SocialIcon,
      },
    });

    const socialIcons = wrapper.findAllComponents(SocialIcon);

    expect(socialIcons.length).toBe(4);
  });
});
