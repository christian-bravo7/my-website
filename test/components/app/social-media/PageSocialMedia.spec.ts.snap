// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSocialMedia component should render correclty 1`] = `<section class="fixed z-40 left-0 grid gap-3 grid-cols-1 page-social-icons"><a target="_blank" href="https://codepen.io/christian-bravo" class="w-6 h-6 md:w-10 md:h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"><svg></svg></a><a target="_blank" href="https://github.com/christian-bravo7" class="w-6 h-6 md:w-10 md:h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"><svg></svg></a><a target="_blank" href="https://www.instagram.com/christian.bravo_/" class="w-6 h-6 md:w-10 md:h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"><svg></svg></a><a target="_blank" href="https://www.linkedin.com/in/christian151-bravo/" class="w-6 h-6 md:w-10 md:h-10 bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-800 hover:text-gray-50 hover:border-pink-800 social-icon rounded-full transition-colors duration-200"><svg></svg></a></section>`;
